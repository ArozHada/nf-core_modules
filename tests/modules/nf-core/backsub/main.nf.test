nextflow_process {

    name "Test Process BACKSUB"
    script "modules/nf-core/backsub/main.nf"
    config "tests/modules/nf-core/backsub/nextflow.config"
    process "BACKSUB"

    test("Background Subtraction Image Markers") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test' ], // meta map
                            file(params.test_data['imaging']['background_subtraction']['image'], checkIfExists: true)
                           ]
                input[1] = [
                            [ id:'test' ], // meta map
                            file(params.test_data['imaging']['background_subtraction']['markers'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
