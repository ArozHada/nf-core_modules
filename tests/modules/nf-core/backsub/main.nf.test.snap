{
    "Background Subtraction Image Markers": {
        "content": [
            [
                "markers_bs.csv:md5,28fddf3b89ed0a9b17b1f8f28e754612",
                "test.backsub.ome.tif:md5,2eff69d894d7ad2509695228f17b6bd1",
                "versions.yml:md5,233a503568378538d3a570d78ea0a670"
            ]
        ],
        "timestamp": "2023-04-07T22:06:38+0000"
    }
}